addons/stock/security/stock_security.xml

```diff
@@ -158,5 +158,12 @@
         <field name="domain_force">['|',('company_id','=',False),('company_id','child_of',[user.company_id.id])]</field>
     </record>
 
+    <record model="ir.rule" id="stock_scrap_company_rule">
+        <field name="name">stock_scrap_company multi-company</field>
+        <field name="model_id" ref="model_stock_scrap"/>
+        <field name="global" eval="True"/>
+        <field name="domain_force">['|',('company_id','=',False),('company_id','child_of',[user.company_id.id])]</field>
```

Comment:
if company_id is requirent, the first part of the rule is useless

Reply:
yes. removed that. Thanks.